### PR TITLE
Check globally excluded URLs in API scan scripts

### LINKED ISSUE #6443: mozilla.net appears in the report even though excluded with globalexcludeurl.url_list.url.regex
**Describe the bug**
Trying to use API scan against and API I get mozilla.net urls in the report, even though it's excluded. Note: Even after running with `-d` I am not able to find anything that leads to mozilla.net, so what I don't understand is why it ends up in the XML report. I also tried a bunch of different regexes and all appear to be ignored.

**To Reproduce**
1. Run the owasp/zap2docker-stable image with `-z "-config globalexcludeurl.url_list.url.regex='https?:\/\/.*mozilla\.net\/.*'"`
2. Wait for it to complete
3. Scroll down to logs and cat report file.
4. See mozilla.net included in zap.log section below

**Expected behavior**
The mozilla.net urls should not be included in the report at all.

**Software versions**
 - ZAP: owasp/zap2docker-stable zap-api-scan.py
 - Add-on: Name and version [e.g. Passive Scan Rules Alpha v19. https://www.zaproxy.org/faq/how-do-i-see-what-version-of-an-add-on-extension-i-have-installed/]
 - OS: Linux/Docker
 - Java: [e.g. openjdk version "1.8.0_171"]
 - Browser: Firefox the Docker image uses

**Errors from the zap.log file**
```
<Snip>
2021-02-06T00:35:29.9852579Z 2021-02-06 00:35:29,981 Starting ZAP
2021-02-06T00:35:29.9854483Z 2021-02-06 00:35:29,981 Params: ['zap-x.sh', '-daemon', '-port', '52443', '-host', '0.0.0.0', '-config', 'database.recoverylog=false', '-config', 'api.disablekey=true', '-config', 'api.addrs.addr.name=.*', '-config', 'api.addrs.addr.regex=true', '-addonupdate', '-addoninstall', 'pscanrulesBeta', '-config', 'rules.cookie.ignorelist=AWSALB,AWSALBCORS', '-config', 'globalexcludeurl.url_list.url.regex=https?:\\/\\/.*mozilla\\.net\\/.*']
2021-02-06T00:35:29.9880505Z 2021-02-06 00:35:29,985 Starting new HTTP connection (1): localhost:52443
2021-02-06T00:35:30.9900804Z 2021-02-06 00:35:30,989 Starting new HTTP connection (1): localhost:52443
<snip>
2021-02-06T00:40:06.8136579Z PASS: Loosely Scoped Cookie [90033]
2021-02-06T00:40:06.8137267Z WARN-NEW: Unexpected Content-Type was returned [100001] x 1 
2021-02-06T00:40:06.8137848Z 	https://REDACTED.com/swagger/index.html (200 OK)
2021-02-06T00:40:06.8138428Z WARN-NEW: Incomplete or No Cache-control and Pragma HTTP Header Set [10015] x 1 
2021-02-06T00:40:06.8139054Z 	https://REDACTED.com/swagger/v1/swagger.json (200 OK)
2021-02-06T00:40:06.8139599Z WARN-NEW: X-Content-Type-Options Header Missing [10021] x 1 
2021-02-06T00:40:06.8140170Z 	https://REDACTED.com/swagger/v1/swagger.json (200 OK)
2021-02-06T00:40:06.8140737Z WARN-NEW: Strict-Transport-Security Header Not Set [10035] x 7 
2021-02-06T00:40:06.8141442Z 	https://REDACTED.com/swagger/v1/swagger.json (200 OK)
2021-02-06T00:40:06.8142226Z 	https://REDACTED.com/accounts/id/quotes?api-version=api-version (401 Unauthorized)
2021-02-06T00:40:06.8143192Z 	https://REDACTED.com/accounts/id?api-version=api-version (401 Unauthorized)
2021-02-06T00:40:06.8143920Z 	https://REDACTED.com/accounts/id?api-version=api-version (401 Unauthorized)
2021-02-06T00:40:06.8144786Z 	https://REDACTED.com/accounts/id/rules?api-version=api-version (401 Unauthorized)
2021-02-06T00:40:06.8145462Z FAIL-NEW: 0	FAIL-INPROG: 0	WARN-NEW: 4	WARN-INPROG: 0	INFO: 0	IGNORE: 0	PASS: 73
2021-02-06T00:40:07.5139399Z <?xml version="1.0"?><OWASPZAPReport version="2.10.0" generated="Sat, 6 Feb 2021 00:40:06">
2021-02-06T00:40:07.5139934Z <site name="https://snippets.cdn.mozilla.net" host="snippets.cdn.mozilla.net" port="443" ssl="true"><alerts><alertitem>
2021-02-06T00:40:07.5140305Z   <pluginid>100001</pluginid>
2021-02-06T00:40:07.5140548Z   <alertRef>100001</alertRef>
2021-02-06T00:40:07.5141326Z   <alert>Unexpected Content-Type was returned</alert>
2021-02-06T00:40:07.5141768Z   <name>Unexpected Content-Type was returned</name>
2021-02-06T00:40:07.5142038Z   <riskcode>1</riskcode>
2021-02-06T00:40:07.5142443Z   <confidence>3</confidence>
2021-02-06T00:40:07.5142882Z   <riskdesc>Low (High)</riskdesc>
2021-02-06T00:40:07.5143749Z   <desc>&lt;p&gt;A Content-Type of text/html was returned by the server.&lt;/p&gt;&lt;p&gt;This is not one of the types expected to be returned by an API.&lt;/p&gt;&lt;p&gt;Raised by the &apos;Alert on Unexpected Content Types&apos; script&lt;/p&gt;</desc>
2021-02-06T00:40:07.5144291Z   <instances>
2021-02-06T00:40:07.5144481Z   <instance>
2021-02-06T00:40:07.5145214Z   <uri>https://snippets.cdn.mozilla.net/6/Firefox/85.0/20210118153634/Linux_x86_64-gcc3/en-US/release-cck-ubuntu/Linux%205.4.0-1039-azure%20(GTK%203.24.20%2Clibpulse%20not-available)/canonical/1.0/</uri>
2021-02-06T00:40:07.5145696Z   <method>GET</method>
2021-02-06T00:40:07.5146114Z   <evidence>text/html</evidence>
2021-02-06T00:40:07.5146333Z   </instance>
2021-02-06T00:40:07.5146525Z   </instances>
```
**Additional context**

**Would you like to help fix this issue?**
Sure, I'd love to.

### PR DESCRIPTION
**Pull request by thc202**
Check if the message being scanned in the HTTP Sender scripts used by
Docker API scan are excluded or not.

Fix #6443.

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

File: docker/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 All notable changes to the docker containers will be documented in this file.
 
+### 2021-02-10
+ - Check if messages being scanned by API scan scripts are globally excluded or not.
```
Comment (kingthorin) [inline docker/CHANGELOG.md:5]:
```suggestion
 - Check if messages being analyzed by API scan scripts are globally excluded or not.
```

Is that safer/clearer?

Comment (kingthorin) [reply docker/CHANGELOG.md:5]: I know we call "passive" scanning, but I wanted it to stay clear that nothing is sent. I guess "passively scanned" would  be fine too.

Comment (thc202) [reply docker/CHANGELOG.md]: Changed.

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks!

**Pull request merged**